traffic_portal/app/src/common/api/DeliveryServiceUrlSigKeysService.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var DeliveryServiceUrlSigKeysService = function(locationUtils, messageModel, $http, $q, ENV) {
+
+	this.generateUrlSigKeys = function(dsXmlId) {
+		var request = $q.defer();
+		$http.post(ENV.api['root'] + 'deliveryservices/xmlId/' + dsXmlId + '/urlkeys/generate')
+		.then(
+			function(result) {
+				messageModel.setMessages([ { level: 'success', text: 'URL Sig Keys generated' } ], true);
```

Comment:
ideally this would probably show result.data.alerts so it can show the message sent back from the server rather than a hardcoded message in the UI. yes, i know, there are lots of instances where the message is hardcoded in the UI. that should be fixed at some point.